### PR TITLE
[test-strategy]Add infra-operator specific rules

### PR DESCRIPTION
**Pull request by gibizer**
This is based on the comment from @stuggi in https://github.com/openstack-k8s-operators/docs/pull/20

### CODE REVIEW

File: test_strategy.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+### nova-operator
+* Each PR changing implementation logic have to include relevant EnvTest
+coverage
+
+### lib-common
+* Adding new functionality needs
+  * either EnvTest coverage if the logic is calling k8s APIs. For example
+  [job](https://github.com/openstack-k8s-operators/lib-common/blob/main/modules/common/test/functional/job_test.go)
+  * of unit test coverage if it isn't calling k8s APIs. For example
```
Comment (stuggi) [inline test_strategy.md:43]: nit 'of' -> 'or'

File: test_strategy.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  [conditions](https://github.com/openstack-k8s-operators/lib-common/blob/main/modules/common/condition/funcs_test.go)
+
+  Except for the `test` and `test-operators` modules as they are test helpers
+  already. Also we don't expect any new functionality added in the `archive`
```
Comment (stuggi) [inline test_strategy.md:47]: I think we are now at a point where we could delete the archive mod. I'll propose a PR for that

**Review by stuggi — COMMENTED** [review]
looks good to me and matches what I thought.

File: test_strategy.md (hunk 2 above)
Comment (gibizer) [reply test_strategy.md]: Thanks. I dropped the archive module from the doc then.

**Review by bogdando — COMMENTED** [review]
lgtm

**Review by stuggi — APPROVED** [review]
/lgtm

**Comment by dprince** [comment]
Not blocking. Just a slight preference that the service operator specific rules be in the respective repositories (lib-common, infra-operator, nova, etc.). I wouldn't look for them in the docs repo myself

**Comment by gibizer** [comment]
> Not blocking. Just a slight preference that the service operator specific rules be in the respective repositories (lib-common, infra-operator, nova, etc.). I wouldn't look for them in the docs repo myself

Eventually I would like this to move there. But for now I would like to have a central place listing our testing rule to:
* to use a dashboard showing where we are with implementing the strategy
* to serve as a sure of peer pressure for other operators to adopt the startegy

**Pull request merged**